api/base/serializers.py

```diff
@@ -857,17 +857,20 @@ def to_representation(self, value):
         self_url = url['self']
         self_meta = self.get_meta_information(self.self_meta, value)
         relationship = format_relationship_links(related_url, self_url, related_meta, self_meta)
-        if related_url and (len(related_path.split('/')) & 1) == 1:
```

Comment:
I'm _guessing_ this used to check for an odd number of segments? Do we know why it was there and why it's safe to remove?

It looks like it was a perhaps hacky way of checking for whether this was a detail endpoint or a list endpoint, but you now need to do work on a detail endpoint. Is that basically the idea?